File: L2ScriptMaker/ToolIncrementer.vb
Public Class ToolIncrementer

    Private Sub StartButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles StartButton.Click

        ResultTextBox.Text = ""

        Dim StartLevel As Double, EndLevel As Double
        Dim Increment As Double
        Dim Variable As Double

        Try
            StartLevel = CDbl(StartLevelTextBox.Text)
            EndLevel = CDbl(EndLevelTextBox.Text)
            If Use1RadioButton.Checked = True Then
                Increment = CDbl(IncValueTextBox.Text)
            Else
                Increment = CDbl(MultValueTextBox.Text)
            End If
            Variable = CDbl(StartValueTextBox.Text)

        Catch ex As Exception
            MessageBox.Show(ex.Message)
            Exit Sub
        End Try

        Dim ParamName As String = ParamNameTextBox.Text
        Dim LeftSym As String = LeftTextBox.Text
        Dim RightSym As String = RightTextBox.Text

        Dim iTemp As Double

        For iTemp = StartLevel To EndLevel
            ResultTextBox.AppendText(ParamName)
            If ParamAddNameCheckBox.Checked = True Then ResultTextBox.AppendText(CStr(iTemp))
            ResultTextBox.AppendText(" = " & LeftSym)
            ResultTextBox.AppendText(Variable.ToString & RightSym & vbNewLine)
            If Use1RadioButton.Checked = True Then
                Variable = Variable + Increment
            Else
                If Variable < 0 Then
                    Increment = -Increment
                Else
                    If Increment < 0 Then
                        Increment = Increment
                    End If
                End If
                Variable = Variable + (Variable * Increment)
            End If
        Next

        TabControl1.SelectedIndex = 1

    End Sub

    Private Sub QuitButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles QuitButton.Click
        Me.Dispose()
    End Sub


    Private Sub Use2RadioButton_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Use2RadioButton.CheckedChanged
        If Use2RadioButton.Checked = False Then
            IncValueTextBox.ReadOnly = False
            MultValueTextBox.ReadOnly = True
        Else
            IncValueTextBox.ReadOnly = True
            MultValueTextBox.ReadOnly = False
        End If
    End Sub
End Class

File: L2ScriptMaker/Modules/Others/ToolIncrementer.cs
using L2ScriptMaker.Extensions.VbCompatibleHelper;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace L2ScriptMaker.Modules.Others
{
	public partial class ToolIncrementer : Form
	{
		public ToolIncrementer()
		{
			InitializeComponent();
		}

		private void StartButton_Click(object sender, EventArgs e)
		{
			ResultTextBox.Text = "";
			double StartLevel;
			double EndLevel;
			double Increment;
			double Variable;

			try
			{
				StartLevel = Conversions.ToDouble(StartLevelTextBox.Text);
				EndLevel = Conversions.ToDouble(EndLevelTextBox.Text);
				if (Use1RadioButton.Checked == true)
					Increment = Conversions.ToDouble(IncValueTextBox.Text);
				else
					Increment = Conversions.ToDouble(MultValueTextBox.Text);
				Variable = Conversions.ToDouble(StartValueTextBox.Text);
			}
			catch (Exception ex)
			{
				MessageBox.Show(ex.Message);
				return;
			}

			string ParamName = ParamNameTextBox.Text;
			string LeftSym = LeftTextBox.Text;
			string RightSym = RightTextBox.Text;

			double iTemp;
			var loopTo = EndLevel;
			for (iTemp = StartLevel; iTemp <= loopTo; iTemp++)
			{
				ResultTextBox.AppendText(ParamName);
				if (ParamAddNameCheckBox.Checked == true)
					ResultTextBox.AppendText(Conversions.ToString(iTemp));
				ResultTextBox.AppendText(" = " + LeftSym);
				ResultTextBox.AppendText(Variable.ToString() + RightSym + Constants.vbNewLine);
				if (Use1RadioButton.Checked == true)
					Variable = Variable + Increment;
				else
				{
					if (Variable < 0)
						Increment = -Increment;
					else if (Increment < 0)
						Increment = Increment;
					Variable = Variable + Variable * Increment;
				}
			}

			TabControl1.SelectedIndex = 1;
		}

		private void QuitButton_Click(object sender, EventArgs e)
		{
			this.Dispose();
		}


		private void Use2RadioButton_CheckedChanged(object sender, EventArgs e)
		{
			if (Use2RadioButton.Checked == false)
			{
				IncValueTextBox.ReadOnly = false;
				MultValueTextBox.ReadOnly = true;
			}
			else
			{
				IncValueTextBox.ReadOnly = true;
				MultValueTextBox.ReadOnly = false;
			}
		}
	}
}
